### DISCUSSION TITLE
Edges and Handles are really hard to click and connect

### DISCUSSION THREAD

**Discussion by ghiscoding** [bcakmakoglu/vue-flow#404] [Q&A] [answered]
Hello, thanks again for helping in previous discussion #386, I have however problems with 2 other things which are
1. really hard to click on the edge (wires) in order to delete them, nearly impossible to click on an animated edge. Would it be possible to somehow increase clickable area without widening the edge?
2. connecting handles sometime just doesn't work and I don't understand why even though I do have the `connectable` enabled and there's no validation in place. The best example of this is the [Hidden/Show](https://vueflow.dev/examples/hidden.html) demo, why can't we reconnect them? However if I go to the [Horizontal](https://vueflow.dev/examples/horizontal.html) demo, it connects without any problems. 

both issues can be seen in the animated gif below from the [Hidden/Show](https://vueflow.dev/examples/hidden.html) demo

These 2 issues are really blocker for us to decide on using the lib or not, I like the lib in general but for our use case, we will connect/reconnect edges and handles a lot so it has to be a good UX to be accepted 😃 

![msedge_IyCu7HGHPm](https://user-images.githubusercontent.com/643976/200404623-bc670866-0ec5-469e-b60b-8e345139f81d.gif)

**Comment by bcakmakoglu** [comment, marked as answer]
Well, you can solve these two issues very easily.

1. Use Custom Edges and add an invisible path with an increased size over your visible one.

```vue
<template>
	<path :id="id" :style="style" class="vue-flow__edge-path" :d="path[0]" :marker-end="markerEnd" />

    <path style="opacity: 0" stroke-width="16" fill="none" :d="path[0]" />
</template>
```

Now you got an increased interaction radius for your edge.

[You can find an example on how to create a custom edge here.](https://vueflow.dev/examples/edges/)

2. Closely check the example code.

The Hidden example has no handler for connecting nodes. Therefore it does not connect them.

**Connections are not made by default**, you have to handle them( *or* set the `autoConnect` option to `true`).

Simplest method to handle connections is this:
```ts
const { onConnect, addEdges } = useVueFlow()

const elements = ref(initialElements)

onConnect((params) => addEdges([params]))
```

If you check the examples again, you will notice that the Horizontal example does in fact have this exact code, that's why connections are created in that one :)

**Reply by bcakmakoglu** [reply]
[Here's an example of increasing the edge interaction radius.](https://codesandbox.io/s/jolly-paper-44xzld?file=/src/CustomEdge.vue)

**Reply by ghiscoding** [reply]
thanks, that is again very useful, however it doesn't seem to show up on my side, do we just need to add `type: 'custom'` to see it rendered with the custom code? I also have this error with TypeScript even if it seems to be exported by the lib (I created a new Vue3/Vite3 project few days ago with default TypeScript settings).

https://github.com/bcakmakoglu/vue-flow/blob/113fc7360a64ab18060cd3a10febdb8405ac7c87/packages/core/src/components/Edges/index.ts#L10

https://github.com/bcakmakoglu/vue-flow/blob/113fc7360a64ab18060cd3a10febdb8405ac7c87/packages/core/src/index.ts#L10-L19

![image](https://user-images.githubusercontent.com/643976/200425589-dc83c7e3-cbad-4907-aa9a-4b6718f2577c.png)

it doesn't show up in the list of available imports
![image](https://user-images.githubusercontent.com/643976/200425860-68bb05a6-330f-4f42-8839-4a10155f1bf4.png)

![image](https://user-images.githubusercontent.com/643976/200427471-61080355-9571-4a1d-8ded-8a66cfa26551.png)


As for the `onConnect`, I did see them in the Examples, but it wasn't clear that it was required for it to connect, I thought they were just event handlers, I didn't realize that it was necessary. Perhaps more emphasis in the docs could be useful 😉

**Reply by bcakmakoglu** [reply]
I'm not sure what you mean "with the custom node"?

To use a custom edge you simply assign it a type like `custom` or whatever name you want to use.
Then you can use the template slot as you showed, using `edge-${nameYouPicked}`.

[There is a guide section on this.](https://vueflow.dev/guide/edge.html#custom-edges)

As for the `EdgeLabelRenderer` - update to at least `1.3.0` to use it (see [CHANGELOG](https://vueflow.dev/changelog/#usage-2))

**Reply by ghiscoding** [reply]
Yes I saw the getting started docs which has more info and did see that the `type: 'custom'` is what I needed from this [doc](https://vueflow.dev/guide/edge.html#custom-edges) but it doesn't show on my side and I wonder if it's because of that TS error and I did upgrade to latest today but even before the update the error came up for both versions, I'm not sure why. I was able to customize the nodes, but I'm having problem customizing the edges

**Reply by bcakmakoglu** [reply]
Can you show me the code of your custom edge?

**Reply by bcakmakoglu** [reply]
Also, have you tried removing your node_modules and reinstalling? ^^

**Reply by ghiscoding** [reply]
hmm I created a public repo with the code and reinstalled and now the error is gone, however the custom edge is still not being rendered. Here's the GitHub repo: https://github.com/ghiscoding/flow-test

the entry point is in [`src/views/Flow.vue`](https://github.com/ghiscoding/flow-test/blob/master/src/views/Flow.vue)

**Reply by bcakmakoglu** [reply]
Remove the `edgeTypes` object and the `default-edge-options` and you're good to go.

1. The default edge options overwrite your type (this seems to be a bug, I'll fix that so that the type isn't overwritten by defaults; Fix is here: https://github.com/bcakmakoglu/vue-flow/pull/407#issuecomment-1306697225)
2. Either use the `edgeTypes` object **or** template slots. Not both.

**Reply by bcakmakoglu** [reply]
After those 2 small changes you can see your custom edge.
<img width="273" alt="image" src="https://user-images.githubusercontent.com/78412429/200488944-f51d3721-577d-408c-a824-eb1f75048491.png">

**Reply by ghiscoding** [reply]
wow awesome that did the trick, isn't the `:edge-types` recommended though? I saw them in the getting started docs, unless it's not necessary with TypeScript?

**Reply by bcakmakoglu** [reply]
It's unrelated to TypeScript.

The way a custom node or edge is resolved is like this:
1. Check if a component is defined with the custom type name in the node or edge types object
2. Check if a slot was passed with the custom node or edge type name
3. Fallback to default

So either pass an custom node / edge types object or use a template slot. 

Personally, I prefer using slots, for no particular reason other than that it's a more familiar way of doing it.
The object way is more React-ish imo.

**Comment by bcakmakoglu** [comment]
With `1.4.0` a new prop for edges is introduced: `interactionWidth`.

It is set to `20` by default (i.e. `20px`).

Using this prop you can increase the interaction width for edges by simply specifying the width you want.
This either requires you to use default edges *or* use the `BaseEdge` component to create custom edges.

(An example will be added to the docs soon 😅)

Relevant release PR: https://github.com/bcakmakoglu/vue-flow/pull/423
